validator-same-as-directive.js

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright (c) 2016 Digital Bazaar, Inc. All rights reserved.
+ */
+define([], function() {
+
+'use strict';
+
+function register(module) {
+  module.directive('brValidatorSameAs', factory);
+}
+
+/* @ngInject */
+function factory() {
+  return {
+    restrict: 'A',
+    require: 'ngModel',
+    scope: {
+      sameAs: '<brValidatorSameAs'
+    },
+    link: Link
+  };
+
+  function Link(scope, element, attrs, ctrl) {
+    scope.$watch('sameAs', function() {
+      ctrl.$validate();
+    });
+
+    ctrl.$validators.brValidatorSameAs = function(modelValue) {
+      if(scope.sameAs !== modelValue) {
+        return false;
+      }
+    };
```

Comment:
@davidlehn We do not want the `$isEmpty` handling as shown in the example because a validation error would not be raised when the compare field to be populated and the confirm field  is left blank.

Mad your suggested revision, ready for another pass.